lib/Runtime/Library/WabtInterface.cpp

```diff
@@ -0,0 +1,206 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeLibraryPch.h"
+
+#ifdef ENABLE_WABT
+#include "../WasmReader/WasmReaderPch.h"
+
+#include "ast-lexer.h"
+#include "common.h"
+#include "ast.h"
+#include "ast-parser.h"
+#include "binary-writer.h"
+#include "binary-writer-spec.h"
+#include "resolve-names.h"
+#include "validator.h"
+#include "Codex/Utf8Helper.h"
+using namespace wabt;
+
+ArenaAllocator* tempArenaPtr;
+struct utf8Allocator
```

Comment:
i don't really like this, and it seems incorrect e.g. in case of multiple threads running. can we update the utf8 api to be able to use non-static allocator?